lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -135,12 +136,16 @@ def join_bulk_responses(bulk_responses)
     def bulk_send(body_stream)
       params = http_compression ? {:headers => {"Content-Encoding" => "gzip"}} : {}
       # Discard the URL 
+      @metric.increment(:bulk_requests)
+      @metric.increment(:concurrent_bulk_requests)
```

Comment:
> Is ES stuck?
> We can infer this by observing two values today: in vs out count for the ES output being nonzero and having both values stable (not increasing). Across two observations (of reasonable time, say, 10 seconds), if in is higher than out and neither value is growing, then we know ES output is waiting on Elasticsearch.

++ I think there are many ways to see the pipeline is stalled 

> Is ES slow?
> We can infer this by observing the duration_in_millis/out to compute per-event latency through Elasticsearch. If the event latency is higher than expected, then Elasticsearch output is a bottleneck.

My concern here is that by computing values per event while logstash works pretty much by batch may lead to wrong conclusions. I don't think CBC is also a crystal clear metric, but can help understand the pipeline behaviour if the values are consistently 0 (no problems in `out` increases), number of workers (may signify backpressure). I don't think values in between can have any meaning.